dali/operators/image/convolution/laplacian_params.h

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_IMAGE_CONVOLUTION_LAPLACIAN_PARAMS_H_
+#define DALI_OPERATORS_IMAGE_CONVOLUTION_LAPLACIAN_PARAMS_H_
+
+#include <numeric>
+#include <vector>
+
+#include "dali/core/small_vector.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/common.h"
+
+
+namespace dali {
+namespace laplacian {
+
+constexpr static const char *kWindowSizeArgName = "window_size";
+constexpr static const char *scaleArgName = "scale";
+constexpr static const float scaleArgDefault = 1.0f;
+constexpr static const char *normalizeArgName = "normalize";
+constexpr static const bool normalizeArgDefault = false;
+
+// OpenCV uses windows sizes up to 31, but may be run on double precision floats.
+// For reasonable precision with floats, 23 is set as a maximal window size, because it is
+// maximal value for which smoothing window coefficients difference do not exceed 10^6.
+// NB. For smoothing window of size 29, float64 and float32 coefficients start to differ.
+constexpr static const int maxWindowSize = 23;
+constexpr static const int defaultWindowSize = 3;
+
+template <typename T>
+class LaplacianWindows {
+ public:
+  LaplacianWindows()
+      : smooth_computed_{1}, deriv_computed_{1}, smoothing_memory_{}, deriv_memory_{} {
+    int offset = 0;
+    int window_size = 1;
+    for (int i = 0; i < num_windows_; i++) {
+      smoothing_views_[i] = {&smoothing_memory_[offset], {window_size}};
+      deriv_views_[i] = {&deriv_memory_[offset], {window_size}};
+      offset += window_size;
+      window_size += 2;
+    }
+    *smoothing_views_[0](0) = 1;
+    *deriv_views_[0](0) = 1;
+  }
+
+  TensorView<StorageCPU, const T, 1> GetWindow(int window_size, bool is_deriv) {
+    assert(1 <= window_size && window_size <= maxWindowSize);
+    assert(window_size % 2 == 1);
+    auto window_idx = window_size / 2;
+    if (!is_deriv) {
+      PrepareSmoothingWindow(window_size);
+      return smoothing_views_[window_idx];
+    }
+    PrepareSmoothingWindow(window_size - 2);
+    PrepareDerivWindow(window_size);
+    return deriv_views_[window_idx];
```

Comment:
It is just my personal preference to format `either this or that` in this way. IMHO early return makes more sense when it is truly early.

```suggestion
    if (!is_deriv) {
      PrepareSmoothingWindow(window_size);
      return smoothing_views_[window_idx];
    } else {
      PrepareSmoothingWindow(window_size - 2);
      PrepareDerivWindow(window_size);
      return deriv_views_[window_idx];
    }
```

Reply:
done